### ISSUE TITLE
Accessibility: Difficulty posting comment to mentor with Orca

### ISSUE THREAD

**Issue by serezlan** [exercism/exercism#5782]
Hi,

Recently I want to post comment to my mentor and unable to do so using Orca screen reader.
I can get into edit field but I cannot review what I have typed using arrow keys.
I also  cannot find button to submit comment..

Please let me know if there's anything I can help to address this issue.

SSystem:
- Debian 10
- - Orca 41rc2
- - Latest Chrome and Firefox

**Comment by angelikatyborska** [comment]
Thank you for submitting this issue! I personally haven't used Orca on Linux so the details might be different, but I do test websites using VoiceOver on a mac and I am also unable to add a comment.

I think this issue is related to [another issue where people complained that they cannot leave comments using the keyboard only](https://github.com/exercism/exercism/issues/5615). One of the problems is that the "send" button only appears after mouse clicks but not when you focus the textarea otherwise.

Another accessibility problem with this component is that the textarea element doesn't have any label associated with it, so a screen reader just says "edit text" without any context. The words "Add a comment" which visually look like a label or a placeholder are added to the form element via a `::before` attribute which doesn't introduce any semantic relation to the textarea at all. I believe this `::before` element of the form needs to become a proper label element for the textarea.

@SleeplessByte I would be interested in hearing your opinion too.

@iHiD Do we have control over that markup?

**Comment by iHiD** [comment]
I'll have a look at all this. Sorry for the pain.

One thing to note is that this editor is currently CodeMirror 5, which is not very accessible. We're waiting for EasyMDE to upgrade to CodeMirror 6, which is accessible, and is on their short-term roadmap (this is the same version we use for the main editor). At that stage, the input in general should become better.

However, it's quite possible that we've also got problems in our implementation of this, so we'll take a look. We sadly ran out of time to really test the mentoring UI (and a few other bits) using screenreaders before launch, but it's on our priority list.

@serezlan Please do open any more issues for any other areas that are not working well (major or minor problems). We'll get them all sorted over time 🙂

**Comment by SleeplessByte** [comment]
@serezlan both me and @angelikatyborska care about this a lot (as does @iHiD), so we will solve it. 

Angelika, your findings are correct. Note that _placeholder_ is actually not meant to serve as a _label_ so something like "write a message" should be a label, and not a placeholder. Many many many sites make this mistake.

@iHiD a sr-only submit button is your friend here. I don't know if that's possible, but it would be a quick fix that solves half the issue.

I'm certain Angelika and I can spend time helping you make it better for users like @serezlan. Just shout when all the other high-prio stuff is done :)

**Comment by serezlan** [comment]
Let me know if you need some hand for testing.

Anyway, I found a workaround


- Type comment on another editor, copy and paste it into comment edit field
- - Exit focus mode and find Toggle Preview button. It is right before edit field.
- - Submit comment button should appear in the bottom. In orca, just press b button once on keyboard to jump to it.


The downside is that any edit need to be done by removing whole block of comment, do editing on external editor, and repeat the process above.
